src/app/content/components/Page/PageComponent.tsx

```diff
@@ -67,6 +68,12 @@ export default class PageComponent extends Component<PagePropTypes> {
   }
 
   public async componentDidUpdate(prevProps: PagePropTypes) {
+    const runId = this.getRunId();
+
+    if (prevProps.page !== this.props.page) {
+      this.postProcess();
```

Comment:
why are we adding this here? where was it done before? it looks like this was only called on mount before, i don't understand how math was working on navigation.

also, i see listenersOff being called in `getSnapshotBeforeUpdate` if the page is different, but nothing re-applying it in the previous code. it was working though, so was something applying it? are you going to end up with multiple listeners?